server/app/controllers/PartController.ts

```diff
@@ -0,0 +1,81 @@
+import { IPart } from '../models/PartModel';
+import { PartService } from '../services/PartService';
+import { inject } from 'inversify';
+import {
+  controller,
+  httpPost,
+  results,
+  httpGet,
+  httpDelete,
+  httpPatch,
+  queryParam,
+} from 'inversify-express-utils';
+import TYPES from '../constants/types';
+import { Request } from 'express';
+import { BaseController } from './BaseController';
+
+@controller('/parts')
+export class PartController extends BaseController {
+  constructor(@inject(TYPES.PartService) private partService: PartService) {
+    super();
+  }
+
+  /*
+  @desc        Get one part or all parts
+  @route       GET /parts or /parts?name= or parts?id=
+  @access      Public
+  @param  {string} id part id
+  @param  {string} name part name
+  */
+  @httpGet('/')
+  public async get(
+    @queryParam('id') id: string | undefined,
+    @queryParam('name') name: string | undefined
```

Comment:
```suggestion
    @queryParam('id') id?: string,
    @queryParam('name') name?: string
```

Can you try this? I think it might be cleaner.